lib/parser.js

```diff
@@ -134,23 +134,24 @@ function parseFromUrl(url, fetchOptions = {}, options) {
 async function dereference(parsedJSON, initialFormat, asyncapiYAMLorJSON, options) {
   try {
     return await refParser.dereference(options.path, parsedJSON, {
+      continueOnError: true,
       parse: options.parse,
       resolve: options.resolve,
       dereference: options.dereference,
     });
   } catch (err) {
     throw new ParserError({
       type: 'dereference-error',
-      title: err.message,
+      title: err.errors[0].message,
       parsedJSON,
-      refs: findRefs(parsedJSON, err.originalPath || err.path, options.path, initialFormat, asyncapiYAMLorJSON),
+      refs: findRefs(err.errors, initialFormat, asyncapiYAMLorJSON),
     });
   }
 }
 
 /*
  * In case of circular refs, this function dereferences the spec again to dereference circular dependencies
- * Last step is to discover those and mark properly with information whey they are pointing
+ * Special property is added to the document that indicates it contains circular refs
```

Comment:
not related to this PR, just noticed this description was old and not matching reality